ports/mimxrt/psram.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2025 Paul Stoffregen
+ *                    Dryw Wade
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/mphal.h"
+
+#if MICROPY_HW_ENABLE_PSRAM
+
+// Copied from here:
+// https://github.com/PaulStoffregen/cores/blob/a23cade55c7530cce43c641ec3ddb8b7f83148a8/teensy4/avr/pgmspace.h#L31
+#define FLASHMEM __attribute__((section(".flashmem")))
+
+// These #defines are from here:
+// https://github.com/PaulStoffregen/cores/blob/a23cade55c7530cce43c641ec3ddb8b7f83148a8/teensy4/imxrt.h
+#define CCM_CCGR_ON				3
+#define FLEXSPI_MCR2_CLRLEARNPHASE(x)		((uint32_t)(x<<14))
+#define FLEXSPI_LUTKEY_VALUE			((uint32_t)0x5AF05AF0)
+#define FLEXSPI_LUT_INSTRUCTION(opcode, pads, operand) ((uint32_t)(\
+	(((opcode) & 0x3F) << 10) | (((pads) & 0x03) << 8) | ((operand) & 0xFF)))
+#define FLEXSPI_LUT_OPCODE_CMD_SDR		0x01
```

Comment:
There is already flexspi_flash_config.h which defines these constants and the LUT entry constructor, maybe worded slightly different.  So maybe it's better to switch to these definitions. It's in the context of flash, but the definitions are for flexspi. If the name of the file matters, it could be unified.